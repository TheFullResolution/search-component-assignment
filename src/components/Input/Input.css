.input-container {
  margin: 0 auto;
  max-width: 500px;
  position: relative;
}

.input-element {
  background-position: calc(100% - 8px);
  background-repeat: no-repeat;
  background-size: 12px;
  border: 1px solid rgb(204, 204, 204);
  line-height: 1;
  padding: 8px;
  transition: border 300ms cubic-bezier(0.445, 0.05, 0.55, 0.95);
  white-space: normal;
  width: 100%;
}

.input-element:focus {
  border: 1px solid black;
  outline: 0;
}

.input-reset-container {
  height: 100%;
  position: absolute;
  right: 28px;
  top: 0;
}

.input-reset {
  background-color: inherit;
  border: none;
  display: block;
  height: 100%;
}

.input-reset img {
  height: 12px;
  vertical-align: middle;
}
