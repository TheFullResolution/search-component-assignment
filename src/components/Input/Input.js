// @flow
import './Input.css'
import React from 'react'

type Props = {}

export const Input = (props: Props) => (
  <div className="input-container">
    <input
      aria-label="Zoeken"
      className="input-element"
      placeholder="Zoeken"
      type="text"
    />
  </div>
)
