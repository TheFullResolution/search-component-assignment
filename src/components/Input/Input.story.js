//@flow

import React from 'react'
import { storiesOf } from '@storybook/react'
import { Input } from './Input.js'

const props = {
  getInputProps: params => params,
  inputValue: 'test',
  reset: () => {}
}

storiesOf('Input', module).add('default', () => <Input {...props} />)
