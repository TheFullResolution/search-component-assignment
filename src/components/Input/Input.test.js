// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { Input } from './Input'

const props = {
  getInputProps: params => params,
  inputValue: 'test',
  reset: () => {}
}

it('Input: default', () => {
  const component = renderer.create(<Input {...props} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
