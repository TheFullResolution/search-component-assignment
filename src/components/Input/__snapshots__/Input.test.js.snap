// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input: default 1`] = `
<div
  className="input-container"
>
  <input
    aria-label="Zoeken"
    className="input-element"
    placeholder="Zoeken"
    style={
      Object {
        "backgroundImage": "url(/images/magnifying-glass.svg)",
      }
    }
  />
  <div
    className="input-reset-container"
  >
    <button
      aria-label="Reset"
      className="input-reset"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Reset"
        src="/images/cancel.svg"
      />
    </button>
  </div>
</div>
`;
